### ISSUE TITLE
Test NG failed to run Parallel tests in different browsers on Sauce labs cloud

### ISSUE THREAD

**Issue by sasank1985** [cbeust/testng#1795]
### TestNG Version
6.9.11

### Expected behavior
Test should run on different browsers at Sauce labs cloud in 2 sessions with provided parameters for each test.

### Actual behavior
Both the tests ran on same browser in different sauce labs sessions when run tests in parallel.

### Is the issue reproductible on runner?

I worked in Eclipse IDE by invoking the sauce labs session by code.

### Test case sample

```java
package ParallelRun;

import java.net.URL;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.openqa.selenium.remote.RemoteWebDriver;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Parameters;
import org.testng.annotations.Test;

public class CrossbrowserParallel {
    public static WebDriver driver;
    public static final String USERNAME = "User1";
    public static final String ACCESS_KEY = "User 1 Access key";
    public static final String SauceURL = "https://" + USERNAME + ":" + ACCESS_KEY + "@ondemand.saucelabs.com:443/wd/hub";
    public static DesiredCapabilities caps;

    @Test
    public void GooglePageLoad() {
        driver.get("https://www.google.co.in");
        driver.manage().window().maximize();
    }

    @Parameters("browsername")
    @BeforeTest
    public void InvokeBrowser(String browsername) throws Exception {
        if (browsername.equalsIgnoreCase("chrome")) {
            caps = DesiredCapabilities.chrome();
            caps.setCapability("platform", "Windows 10");
        } else if (browsername.equalsIgnoreCase("firefox")) {
            caps = DesiredCapabilities.firefox();
            caps.setCapability("platform", "Windows 10");
        } else if (browsername.equalsIgnoreCase("ie")) {
            caps = DesiredCapabilities.internetExplorer();
            caps.setCapability("platform", "Windows 10");
        } else if (browsername.equalsIgnoreCase("safari")) {
            caps = DesiredCapabilities.safari();
            caps.setCapability("platform", "OS X 10.11");
        }
        caps.setCapability("name", "Excelicare Login cross browser testing-" + browsername);
        caps.setCapability("tags", "ECv6.8.0.1");
        caps.setCapability("build", "Excelicare");
        driver = new RemoteWebDriver(new URL(SauceURL), caps);
    }

    @AfterTest
    public void BrowserClose() throws Exception {
        driver.quit();
    }
}
```

Test NG  XML Used 

```xml 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="Suite" parallel="tests">
    <test name="ChromeTest">
        <parameter name="browsername" value="chrome"/>
        <classes>
            <class name="ParallelRun.CrossbrowserParallel"/>
        </classes>
    </test>
    <test name="FirefoxTest">
        <parameter name="browsername" value="firefox"/>
        <classes>
            <class name="ParallelRun.CrossbrowserParallel"/>
        </classes>
    </test>
    <test name="IETest">
        <parameter name="browsername" value="ie"/>
        <classes>
            <class name="com.excelicare.saucelabs.jenkins.ECLoginCrossbrowserParallel"/>
        </classes>
    </test>
</suite>

**Comment by sasank1985** [comment]
![xml screenshot](https://user-images.githubusercontent.com/22906393/40236034-d36e916e-5ac9-11e8-8621-b81586ff1ab5.jpg)

**Comment by krmahadevan** [comment]
The problem is not with TestNG, but its with your test code. Your `WebDriver` and your `DesiredCapabilities` objects are static in nature. This means that when TestNG runs your class `ParallelRun.CrossBrowserParallel` in parallel, both of your `@BeforeTest` methods are basically trying to initialize the same object. So the one that runs the last, basically sets the WebDriver flavor. Please fix your test code and remove static references on WebDriver and DesiredCapabilities object.

PS: You are using a very old version of TestNG. The latest released version of TestNG is `6.14.3`
